Skeleton/src/Volunteer.cpp

```diff
@@ -1 +1,263 @@
+        timeLeft = timeLeft - 1;
+    }
+    return (timeLeft == 0);
+}
+
+bool CollectorVolunteer::hasOrdersLeft() const  // Signal whether the volunteer didn't reach orders limit,Always true for CollectorVolunteer and DriverVolunteer
+{
+    return true;
+} 
+
+bool CollectorVolunteer::canTakeOrder(const Order &order) const 
+{
+    return (order.getStatus() == OrderStatus::PENDING && !Volunteer::isBusy());
+}
+
+void CollectorVolunteer::acceptOrder(const Order &order) 
+{
+    if(canTakeOrder(order)){
+        activeOrderId = order.getId();
+        timeLeft = coolDown;
+    }
+}
+string CollectorVolunteer::toString() const 
+{
+    return "Collector Volunteer - ID: " + std::to_string(getId())
+     + ", Name: " + getName() 
+     + ", Active Order ID: " + std::to_string(getActiveOrderId()) 
+     + ", Completed Order ID: " + std::to_string(getCompletedOrderId());
+}
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+//////////////////////////// LimitedCollector Volunteer //////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+
+LimitedCollectorVolunteer::LimitedCollectorVolunteer(int id, const string &name, int coolDown ,int maxOrders) : CollectorVolunteer(id, name, coolDown), maxOrders(maxOrders), ordersLeft(maxOrders) {}
+LimitedCollectorVolunteer* LimitedCollectorVolunteer::clone() const 
+{
+    return new LimitedCollectorVolunteer(*this);
+}
+
+bool LimitedCollectorVolunteer::hasOrdersLeft() const 
+{
+    return (ordersLeft > 0);
+}
+
+bool LimitedCollectorVolunteer::canTakeOrder(const Order &order) const 
+{
+    return (order.getStatus() == OrderStatus::PENDING && !CollectorVolunteer::isBusy() && hasOrdersLeft());
+}
+
+void LimitedCollectorVolunteer::acceptOrder(const Order &order) 
+{
+    if(canTakeOrder(order)){
+        ordersLeft--;
+        activeOrderId = order.getId();
+    }
+}
+
+int LimitedCollectorVolunteer::getMaxOrders() const 
+{
+    return maxOrders;
+}
+int LimitedCollectorVolunteer::getNumOrdersLeft() const
+{
+    return ordersLeft;
+}
+string LimitedCollectorVolunteer::toString() const 
+{
+        return "Limited Collector Volunteer - ID: " + std::to_string(getId()) +
+           ", Name: " + getName() + ", Active Order ID: " + std::to_string(getActiveOrderId()) +
+           ", Completed Order ID: " + std::to_string(getCompletedOrderId()) +
+           ", Orders Left: " + std::to_string(getNumOrdersLeft());
+}
+
+////////////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Driver Collector ///////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////
+
+DriverVolunteer::DriverVolunteer(int id, const string &name, int maxDistance, int distancePerStep) : Volunteer(id,name), maxDistance(maxDistance), distancePerStep(distancePerStep), distanceLeft(NO_ORDER) {}
```

Comment:
i think distance left 0 bo NO_ORDER

Reply:
I think you are right, because 'acceptOrder' will be called before decreaseDistanceLeft(), right?

Reply:
yes